subprojects/packagefiles/pcre2/meson.build

```diff
@@ -208,12 +208,30 @@ if host_machine.system() == 'windows'
   general_c_args += ['-D_CRT_SECURE_NO_DEPRECATE', '-D_CRT_SECURE_NO_WARNINGS']
 endif
 
+have_version_scripts = c_compiler.has_link_argument(
+  '-Wl,--version-script,@0@/src/libpcre2-8.sym'.format(
+    meson.current_source_dir(),
+  ),
+)
+foreach lib : ['pcre2-8', 'pcre2-posix', 'pcre2-16', 'pcre2-32']
+  set_variable(
+    lib.underscorify() + '_link_args',
+    have_version_scripts ? [
```

Comment:
Could use https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerget_supported_link_arguments

Not sure if worth it.